### ISSUE TITLE
LightGBM produce the same prediction even the other feature values are different except 1st feature

### ISSUE THREAD

**Issue by hashithacd** [microsoft/LightGBM#4647]
I have trained a lightgbm model with basic hyperparameters and model is reasonably accurate after the fine tuning. 

deployed the model in production ( save the model as a pickle) and when predicting, it produce the same probability value even the different values for other features. if i change the first feature only the model produce different probabilities.

note that all the features are categorical features and response variable is a binary.

@guolinke @chivee @wxchan @StrikerRUS please help me on this.

**Comment by shiyu1994** [comment]
@hashithacd Thanks for using LightGBM. Could you please check whether the produced model uses the first feature only? You can save the model into a file and check the split features in the tree structures. For example, in python, you may use the `save_model` method, see
https://lightgbm.readthedocs.io/en/latest/pythonapi/lightgbm.Booster.html#lightgbm.Booster.save_model
for reference.

**Comment by hashithacd** [comment]
@shiyu1994 thank you for the quick turnaround. 
yes i did the same as you mentioned. it shows all the features that are intended. 

would you think any other reason behind this issue?

**Comment by shiyu1994** [comment]
@hashithacd In that case, could you please provide a reproducible example? That would be very helpful for us to identify the problem.

**Comment by hashithacd** [comment]
@shiyu1994 @jameslamb  Sorry for the delay in response. 
As an example;
	I have 5 features and response variable. All 5 features are categorical features with high cordiality (1500+ categories per feature). I used the categorical_feature parameter in lightgbm to handle the categorical features.
	 Let’s consider V1, V2, …, V5 are the features 
Fit the model and predict for a sample using predict_proba function
Parameters = {
               'boosting_type': 'gbdt',    'objective': 'binary',    'importance_type': 'gain',    'n_estimators': 5000,
 'max_depth': 10,    'num_leaves': 32,    'learning_rate': 0.01,    'is_unbalance': True, 'min_data_per_group': 1, }

	a1.V1 , a2.V2, a3.V3, a4.V4, a5.V5   return the prediction as 0 with probability of 0.78
	then changing the a1 will be changed the prediction 
	But, changing any other values of “a(i)” does not change the prediction and it is the same as 0 with probability of 0.78
 Hope this will help you to understand the issue.
Note that, since those are sensitive data, I am unable to give the actual sample data here.

**Comment by jameslamb** [comment]
Thanks @hashithacd . Unfortunately, **we cannot provide additional help without a reproducible example**.

There are a lot of possibilities for how the behavior you're observed could be happening, and without a reproducible example maintainers would have have to spend time guessing at what is going on.

Here are all the things I can think of, off the top of my head, that could lead to an issue like the one you've described:

* the distribution of the features being presented to your model "in production" is very different from the distribution observed during training, such that you're getting "locked in" to a small subset of paths through the trees
* you are using a different operating system / LightGBM version / Python version / version of some other Python dependency "in production" (compared to what you used in training, when creating that model object) and that is leading to issues like loss of numerical precision or some other issue
* the model you've trained only has splits on `a1` (I know you said you think this isn't happening, but you haven't presented evidence of that yet)
* the model you've trained always uses `a1` for its FIRST split
* the changes you're attempting for other features besides `a1` are not large enough to cross bin boundaries and result in different paths through the trees
